src/models/model.cpp

```diff
@@ -533,6 +532,15 @@ void Model::CreateSessionOptionsFromConfig(const Config::SessionOptions& config_
         opts.emplace(option.first, option.second);
       }
       session_options.AppendExecutionProvider("WebGPU", opts);
+    } else if (provider_options.name == "openvino") {
+      std::unordered_map<std::string, std::string> ov_options;
+      for (auto& option : provider_options.options) {
+        ov_options.emplace(option.first, option.second);
+      }      
+      session_options.AppendExecutionProvider("OpenVINO", ov_options);
+      
+      if (is_primary_session_options)
+        p_device_ = GetDeviceInterface(DeviceType::OPENVINO);      
```

Comment:
This PR should be merged soon and will simplify this: https://github.com/microsoft/onnxruntime-genai/pull/1384/files

Also, we need to have the provider_options.name match the ORT provider name, so it should be "OpenVINO" not "openvino". Going forward we'll have them match and the PR above has a backwards compatibility fix for the existing ones.

Reply:
The mentioned PR is now merged, please sync again and merge this with the simplified version.